### ISSUE TITLE
Run Test Content 2022-10-17

### ISSUE THREAD

**Issue by louaybassbouss** [cta-wave/dpctf-tests#96]
We ran the tests with the new test content from 2022-10-17 generated by @rbouqueau. Everything looks good except the following 2 points:

- [ss1 14.985_29.97_59.94](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/stream.mpd):  
Video fails to load:  
GET https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/cfhd_sets/15_30_60/t19/0.m4s 404 (Not Found)

- ss1 zips [15_30_60](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/15_30_60/ss1/2022-10-17/ss1.zip), [12.5_25_50](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/12.5_25_50/ss1/2022-10-17/ss1.zip) and [14.985_29.97_59.94](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/ss1.zip):  
Structure and content of the zip files are wrong  

@jpiesing @yanj-github FYI

**Comment by nicholas-fr** [comment]
There is also this issue https://github.com/cta-wave/Test-Content-Generation/issues/56 related to the ss1 zip structure.

**Comment by rbouqueau** [comment]
> https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/stream.mpd

Where did you get that URL from? I can't find it in [database.json](https://cta-wave.github.io/Test-Content/database.json).

> ss1 zips [15_30_60](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/15_30_60/ss1/2022-10-17/ss1.zip), [12.5_25_50](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/12.5_25_50/ss1/2022-10-17/ss1.zip) and [14.985_29.97_59.94](https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/ss1.zip):
Structure and content of the zip files are wrong

@louaybassbouss Could you detail what is wrong?

I see things like these, which seems correct to me:

```

[...]
  1313959  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/0.m4s
  1578390  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/337920.m4s
  1274753  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/122880.m4s
  1544270  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/276480.m4s
  1424643  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/645120.m4s
  1340101  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/92160.m4s
  1536012  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/368640.m4s
  1342949  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/798720.m4s
  1311558  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/522240.m4s
  1166261  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/491520.m4s
  1445230  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/890880.m4s
  1542120  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/245760.m4s
  1293639  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/768000.m4s
  1549055  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/860160.m4s
  1378333  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/706560.m4s
  1472830  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/1/153600.m4s
     1694  2022-10-17 08:19   chdf_sets/15_30_60/t20/2022-10-17/stream.mpd
[...]
```

> There is also this issue https://github.com/cta-wave/Test-Content-Generation/issues/56 related to the ss1 zip structure.

Ok, not implemented yet. I would plan to do it when the next version is generated.

**Comment by jpiesing** [comment]
> > https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/stream.mpd
> 
> Where did you get that URL from? I can't find it in [database.json](https://cta-wave.github.io/Test-Content/database.json).

@louaybassbouss  Please can you respond to @rbouqueau ? I really want to get this version to Eurofins and then the next iteration made.

**Comment by rbouqueau** [comment]
> I can confirm this structure. I assume the directory to be called ss1 instead of t20, like it was so far. Correct me if I am wrong here, but what would be the difference to the t20 content? Additionally, there is a cfhd_sets directory with several more t* directories. It didn't know that is supposed to be like this. Also, the stream.mpd in the root.

Ok, several things.

There are two WAVE media profiles that we use here: ```cfhd``` and ```chdf```. ```chdf``` only contains ```t20```. We used to not (and not be able) differentiate them, but now we do (https://github.com/cta-wave/Test-Content-Generation/issues/52).

As you may know, the folder structure has changed. So these contents (belonging to two different media profiles) now belongs to unrelated folders.

In addition the spec was updated and now contains some switching set examples. So what is implemented is the latest spec.

```stream.mpd``` name is planned to change, see https://github.com/cta-wave/Test-Content-Generation/issues/56. The zip is supposed to be a zip of the folder that is uploaded so you will be affected again. Please comment there if you have a preference.

**Comment by jpiesing** [comment]
@FritzHeiden @louaybassbouss Ping? Please bring an update on this to the DPCTF meeting later today.

**Comment by FritzHeiden** [comment]
Let me try to rephrase my problem:

Usually, online content and local content have the same structure. For example:

Online:  
```
https://dash.akamaized.net/WAVE/vectors/cfhd_sets/12.5_25_50/t1/2022-10-17/stream.mpd
```

Local:  
```
/cfhd_sets/12.5_25_50/t1/2022-10-17/stream.mpd
```

However, with the ss1 content the online contents path does not match any of the local contents MPDs:

Online:  
```
https://dash.akamaized.net/WAVE/vectors/cfhd_sets/12.5_25_50/ss1/2022-10-17/stream3.mpd
```

Local:  
```
/stream.mpd
/chdf_sets/12.5_25_50/t20/2022-10-17/stream.mpd
```

Neither of the MPDs names match, and for the second MPD there is t20 instead of ss1 in the path. Which MPD should be used?

Additionally, I wanted to point out there are more directories with video files:

```
/cfhd_sets/12.5_25_50/t19
/cfhd_sets/12.5_25_50/t23
/cfhd_sets/12.5_25_50/t24
/cfhd_sets/12.5_25_50/t25
/cfhd_sets/12.5_25_50/t28
/cfhd_sets/12.5_25_50/t32
/cfhd_sets/12.5_25_50/t34
```

I will attend the call later today, so we may discuss any further questions

**Comment by jpiesing** [comment]
@rbouqueau @FritzHeiden hopefully we can resolve this today and start moving forwards again.

**Comment by rbouqueau** [comment]
> However, with the ss1 content the online contents path does not match any of the local contents MPDs

I've removed the reference to ```stream3.mpd``` (from a previous discussion we had) both on the database and the server. From now on, just consider ```stream.mpd```. ```stream3.mpd``` is not needed anymore in the last release.

As for the folder tree... Quite some time ago, I asked if we should either 1) point to the content or 2) copy it. I was answer to copy it (i.e. 2)). Rationale: useful content is 1) not easy to identify and 2) old content could be deemed deprecated (and eventually deleted). So the content is ss1 is a replication of some other content.

I realize all of this is confusing because this release implements a lot of breaking changes and duplicated content. I've tried to stick to the instructions and specifications. Eventually it seems logical to me. So I hope it will seem logical for you too.

> /stream.mpd
> /chdf_sets/12.5_25_50/t20/2022-10-17/stream.mpd

Consider ```/stream.mpd```.

The second mpd is the single stream. This is an artifact of the copy (but the [script](https://github.com/cta-wave/Test-Content-Generation/blob/master/ss1/switching_set_X1.sh) could remove it, just let me know).

> Additionally, I wanted to point out there are more directories with video files [...]

SS1 is the combination of t20 (from the CHDF) and t19,t23,t24,t25,t28,t32,t34 (from the CFHD media profile). I've just checked in the [matrix](https://docs.google.com/spreadsheets/d/1hxbqBdJEEdVIDEkpjZ8f5kvbat_9VGxwFP77AXA_0Ao/edit#gid=0) and it seems ok.

**Comment by rbouqueau** [comment]
MPD filenames updated.

**Comment by rbouqueau** [comment]
I confirm there is an issue in the template MPDs for the switching sets.

**Comment by rbouqueau** [comment]
May I ask you to retry?

**Comment by FritzHeiden** [comment]
I had the following issues:

**Local 14.985_29.97_59.94**:

From zip I used:

```
/avc_14.985_29.97_59.94_ss1_stream.mpd
```

The error I got:

```
Failed to load resource: the server responded with a status of 404 (Not Found)
for resource: http://web-platform.test:8000/content/cfhd_sets/14.985_29.97_59.94/t19/1801800.m4s
```

When having a look inside the zip file, there is only a `/cfhd_sets/14.985_29.97_59.94/t19/180180.m4s` and no `/cfhd_sets/14.985_29.97_59.94/t19/1801800.m4s`

**Online 14.985_29.97_59.94**:

The URL I used:

```
https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/avc_14.985_29.97_59.94_ss1_stream_v3.mpd
```

The error I got:

```
GET https://dash.akamaized.net/WAVE/vectors/cfhd_sets/14.985_29.97_59.94/ss1/2022-10-17/cfhd_sets/14.985_29.97_59.94/t19/1801800.m4s 404 (Not Found)
```

Seems to be the same issue as for the local content

**Comment by rbouqueau** [comment]
I've fixed the issue. Can you retry?

**Comment by rbouqueau** [comment]
I confirm. This was done because the single streams didn't need to be regenerated.

**Comment by jpiesing** [comment]
> I confirm. This was done because the single streams didn't need to be regenerated.

@rbouqueau When will this temporary solution go away?

**Comment by rbouqueau** [comment]
Next time the content is regenerated.

**Comment by jpiesing** [comment]
> Last question for @rbouqueau before I create ticket for Eurofins: the filenames of ss1 content in database.json include version e.g. `ss1_v4.zip` or `avc_12.5_25_50_ss1_stream_v4.mpd`. I understood that this was temporary solution while fixing the issues related to the content. Can you confirm @rbouqueau?

@louay: Did you get an answer sufficient to create a ticket for Eurofins in the OF project? Please could you link it from here.

**Comment by FritzHeiden** [comment]
I created the issue here https://github.com/cta-wave/device-observation-framework/issues/41

**Comment by pshorrock** [comment]
@FritzHeiden before we look to perform any validation of the Observation Framework can you please summarise the testing that you have done. Have all of the latest version of the tests been run with the latest version of the test runner? If so what checks have you performed? If not all of the tests which have you run? Presumably whatever you have done has covered all media streams as a minimum? Have you used the Observation Framework? If not how have you validated what is there is correct? And what devices have you used? TVs? Anything else? We plan to base the bulk of our testing on TVs but also at least a subset on a mobile phone. As this is going to be fairly comprehensive testing we do not want to start until we are sure it is ready for us to validate the OF - and that will be based on what testing has been done to date.

**Comment by FritzHeiden** [comment]
@pshorrock I changed the new content provided by romain for the old one, regenerated all tests and ran them in the browser. What I was looking for was that the tests generate and execute without errors. I did not validate any semantics in the content, like annotations or QR codes. I just checked if the video plays properly (as visible to the human eye).

**Comment by yanj-github** [comment]
Thanks @FritzHeiden can you kindly provide us the following detail please? We would like to install exact same version of test runner before we can start our validation.

- Which version of dpctf-tests/dpctf-test-runner/dpctf-deploy did you use for testing? Can you provide branches and commit hex code or please tag them and let us know the tag name if that is eaiser. 
- I think version of dpctf-tests already covered which test contents that are linked, but if not please confirm the version/url of test contents that was used.
- Which browser and device have you run tests on?
- Can you share the result json file or test report please?

**Comment by louaybassbouss** [comment]
@yanj-github lets move the discussion to this issue https://github.com/cta-wave/device-observation-framework/issues/41 .  @FritzHeiden provided the command you can use to start the test runner. As he mentioned, what is changed is only the content. Test Runner, Deploy, and Test Implementation are not changed. @FritzHeiden will provide the Json reports in this issue [41](https://github.com/cta-wave/device-observation-framework/issues/41)
